From 0da6e533679c177c1f5efffd22e345304d273e49 Mon Sep 17 00:00:00 2001
From: Oliver Winker <oliver@oli1170.net>
Date: Sun, 21 Aug 2011 21:20:32 +0200
Subject: [PATCH] common, usb: Include kernel version number "2.6.40" in 3.0.0 changes

Fedora e.g. works with kernel version number "2.6.40" for 3.0.0 kernels
for transitional purposes.

Include therefore explicitly kernel version number "2.6.40"
in the 3.0.0 adaptation code (logically one check on >=2.6.40 would have
been sufficient, but reading-wise it's then maybe less obvious, that it's
about 3.0.0 adaptation code).

Signed-off-by: Oliver Winker <oliver@oli1170.net>
---
 common.c |    3 ++-
 usb.c    |    3 ++-
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/common.c b/common.c
index bdd7de4..363efb9 100644
--- a/common.c
+++ b/common.c
@@ -31,7 +31,8 @@
 #include <linux/ethtool.h>
 #include <linux/nl80211.h>
 #include <linux/ieee80211.h>
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 40)
 #include <linux/ratelimit.h>
 #endif
 
diff --git a/usb.c b/usb.c
index 54cf450..8022bd0 100644
--- a/usb.c
+++ b/usb.c
@@ -42,7 +42,8 @@
 #include <linux/ethtool.h>
 #include <linux/workqueue.h>
 #include <linux/nl80211.h>
-#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0)
+
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(3, 0, 0) || LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 40)
 #include <linux/ratelimit.h>
 #endif
 
-- 
1.7.4.1

